src/smt/smt_engine.cpp

```diff
@@ -942,7 +977,17 @@ LogicInfo SmtEngine::getUserLogicInfo() const
   res.lock();
   return res;
 }
-void SmtEngine::setFilename(std::string filename) { d_filename = filename; }
+
+void SmtEngine::setFilename(std::string filename)
+{
+  d_filename = filename;
+
+  // Copy the original options. This is called prior to beginning parsing.
+  // Hence reset should revert to these options, which note is after reading
+  // the command line.
+  d_originalOptions.copyValues(d_options);
```

Comment:
Why is this done in `setFilename`?

Reply:
I could change the name of this method to `notifyStartParsing`? Regardless it will probably be the case that this is refactored soon.

Reply:
I'd suggest renaming it.